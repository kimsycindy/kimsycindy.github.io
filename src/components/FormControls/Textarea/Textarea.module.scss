@import '../../../styles/variables.scss';

.textarea {
  background-color: $gray-light;
  padding: 10px;
  line-height: $lineHeight-regular;
  width: 100%;
}
