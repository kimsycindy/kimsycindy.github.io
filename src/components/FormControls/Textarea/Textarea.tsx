import React from 'react'

import styles from './Textarea.module.scss'

interface Props {
  onChange: (e: React.ChangeEvent<HTMLTextAreaElement>) => void
  name: string
  label: string
  value: string
  placeholder?: string
  required?: boolean
}

export const Textarea: React.FC<Props> = ({
  onChange,
  name,
  label,
  value,
  placeholder,
  required = false,
}) => (
  <div>
    <label htmlFor={name}>
      {label}
      <br />
      <textarea
        className={styles.textarea}
        onChange={onChange}
        name={name}
        value={value}
        placeholder={placeholder}
        required={required}
        rows={5}
        cols={33}
      />
    </label>
  </div>
)

export type FormControlsTextareaElement = HTMLTextAreaElement
