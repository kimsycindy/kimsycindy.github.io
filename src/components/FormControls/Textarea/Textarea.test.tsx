import React from 'react'
import renderer from 'react-test-renderer'

import { Textarea } from './Textarea'

describe('FormControls Textarea', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Textarea
          onChange={jest.fn()}
          name="mock"
          label="Mock"
          value=""
          placeholder="Placeholder"
          required
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
