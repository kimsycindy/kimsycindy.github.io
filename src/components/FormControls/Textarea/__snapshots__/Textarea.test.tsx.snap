// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControls Textarea renders correctly 1`] = `
<div>
  <label
    htmlFor="mock"
  >
    Mock
    <br />
    <textarea
      className="textarea"
      cols={33}
      name="mock"
      onChange={[MockFunction]}
      placeholder="Placeholder"
      required={true}
      rows={5}
      value=""
    />
  </label>
</div>
`;
